docs/operations/metrics.md

```diff
@@ -264,40 +264,42 @@ If the JVM does not support CPU time measurement for the current thread, `ingest
+| `worker/task/failed/count`               | Number of failed tasks run on the reporting worker per emission period. This metric is only available if the `WorkerTaskCountStatsMonitor` module is included, and is only supported for Middle Manager nodes.     | `category`, `workerVersion`|Varies|
+| `worker/task/success/count`              | Number of successful tasks run on the reporting worker per emission period. This metric is only available if the `WorkerTaskCountStatsMonitor` module is included, and is only supported for Middle Manager nodes. | `category`,`workerVersion`|Varies|
+| `worker/taskSlot/idle/count`             | Number of idle task slots on the reporting worker per emission period. This metric is only available if the `WorkerTaskCountStatsMonitor` module is included, and is only supported for Middle Manager nodes.      | `category`, `workerVersion`|Varies|
+| `worker/taskSlot/total/count`            | Number of total task slots on the reporting worker per emission period. This metric is only available if the `WorkerTaskCountStatsMonitor` module is included.                                                     | `category`, `workerVersion`|Varies|
+| `worker/taskSlot/used/count`             | Number of busy task slots on the reporting worker per emission period. This metric is only available if the `WorkerTaskCountStatsMonitor` module is included.                                                      | `category`, `workerVersion`|Varies|
+| `worker/task/assigned/count`             | Number of tasks assigned to an indexer per emission period. This metric is only available if the `WorkerTaskCountStatsMonitor` module is included.                                                                 |`dataSource`|Varies|
+| `worker/task/completed/count`            | Number of tasks completed by an indexer per emission period. This metric is only available if the `WorkerTaskCountStatsMonitor` module is included.                                                                |`dataSource`|Varies|
+| `worker/task/running/count`              | Number of tasks running on an indexer per emission period. This metric is only available if the `WorkerTaskCountStatsMonitor` module is included.                                                                  |`dataSource`|Varies|
+| `s3upload/threadPool/taskQueuedDuration` | Milliseconds spent by a task in queue before it starts uploading chunk to S3 when durable storage is enabled.                                                                                                      ||Varies|
+| `s3upload/threadPool/queuedTasks`        | The number of tasks that are currently queued and waiting to upload chunks to S3 when durable storage is enabled.                                                                                                  ||Varies|
```

Comment:
I am not sure but these metrics don't seem particularly useful. A more meaningful metric would be time taken to upload a single chunk, and total time to upload all chunks of a single output.

You may use `uploadId` or some other equivalent field as a dimension in the second metric.

Also, it doesn't seem we ever log the `uploadId` and/or upload path in `RetryableS3OutputStream`. That could be helpful too.

Reply:
> Also, it doesn't seem we ever log the uploadId and/or upload path in RetryableS3OutputStream. That could be helpful too.

@kfaraz Karan suggested (in my previous PR) that it would be too noisy to log it, so I removed it.

Reply:
@kfaraz Have added a metric for individual task duration, with dimensions `uploadId` and `partNumber` - these 2 dimensions uniquely identify any part.
Hope that works!

Reply:
> @kfaraz Karan suggested (in my previous PR) that it would be too noisy to log it, so I removed it.

There is already an info log line in `RetryableS3OutputStream.close()`, line 213. You can just include the upload ID and path in that message.

Reply:
> @kfaraz Have added a metric for individual task duration, with dimensions uploadId and partNumber - these 2 dimensions uniquely identify any part.
Hope that works!

Thanks!

Reply:
> There is already an info log line in RetryableS3OutputStream.close(), line 213. You can just include the upload ID and path in that message.

Done